python/common/org/python/Object.java

```diff
@@ -32,6 +32,11 @@
     public org.python.Object byValue();
 
     /**
+     * Perform the appropriate reflected comparison operation
+     */
+    public org.python.Object comparison(org.python.Object other, String operator);
```

Comment:
For historical reasons, and to preserve the `__`-method-means-internals metaphor, this should probably be called `__cmp__`. Strictly, Python3 doesn't need a __cmp__ operator, but if we're going to introduce one for our own purposes, it wouldn't hurt to retain consistency.